server/src/main/java/org/tctalent/server/service/db/impl/TaskAssigmentServiceImpl.java

```diff
@@ -84,6 +88,9 @@ public TaskAssignmentImpl assignTaskToCandidate(
 
             default:
                 taskAssignment = new TaskAssignmentImpl();
+                if (task.getName().equals("duolingoTest")) {
+                    couponService.assignCouponToCandidate(candidate.getId());
+                }
```

Comment:
This creates a dependency between TaskAssignment and Coupons. See my comments on the SavedListServiceImpl above. 

I think a better approach which keeps services much more loosely coupled is to do two API calls when a button is pressed - one which assigns the coupon(s) and one which assigns the tasks.

Alternately - do it all through the DuoLingoCouponAdminApi. When a coupon is assigned, a task is created notifying the candidate(s).

It makes more sense for the DuoLingoCouponService to depend on the TaskService than to have the TaskService depend on the DuoLingoCouponService.